File: bjo14888.py
N=int(input())
num_list=list(map(int, input().split()))
add, sub, mul, div=map(int, input().split())
max_num, min_num=-99999, 99999

def solution(num, result, add, sub, mul, div):
    global max_num, min_num
    if num==N:
        max_num=max(result, max_num)
        min_num=min(result, min_num)
        return 

    else:
        if add:
            solution(num+1, result+num_list[num], add-1, sub, mul, div)
        if sub:
            solution(num+1, result-num_list[num], add, sub-1, mul, div)
        if mul:
            solution(num+1, result*num_list[num], add, sub, mul-1, div)
        if div:
            solution(num+1, int(result/num_list[num]), add, sub, mul, div-1)

solution(1, num_list[0], add, sub, mul, div)
print(max_num)
print(min_num)

File: bjo14888.cpp
#include <iostream>
using namespace std;

int N;
int num_list[11];
int operators[4];
int max_num = -99999;
int min_num = 99999;

void solution(int result, int idx) {
	if (idx == N) {
		if (result > max_num)
			max_num = result;
		if (result < min_num)
			min_num = result;
		return;
	}
	for (int i = 0; i < 4; i++) {
		if (operators[i] > 0) {
			operators[i]--;
			if (i == 0)
				solution(result + num_list[idx], idx + 1);
			else if (i == 1)
				solution(result - num_list[idx], idx + 1);
			else if (i == 2)
				solution(result * num_list[idx], idx + 1);
			else
				solution(result / num_list[idx], idx + 1);
			operators[i]++;
		}
	}
	return;
}

int main() {
	cin >> N;
	for (int i = 0; i < N; i++)
		cin >> num_list[i];
	for (int i = 0; i < 4; i++)
		cin >> operators[i];
	solution(num_list[0], 1);
	cout << max_num << endl;
	cout << min_num;
	system("pause");
}